tests/integration/test_dev_server_serve.py

```diff
@@ -9,11 +14,55 @@
 from .dev_server_test_helper import MyFakeSite, SERVER_ADDRESS, find_unused_port, LOGGER, OUTPUT_FOLDER
 
 
+def test_server_on_used_port(site_and_base_path: Tuple[MyFakeSite, str]):
+    site, base_path = site_and_base_path
+    command_serve = serve.CommandServe()
+    command_serve.set_site(site)
+    command_serve.serve_pidfile = "there is no file with this name we hope"
+    command_serve.logger = logging.getLogger("dev_server_test")
+    catch_log = StringIO()
+    catch_log_handler = logging.StreamHandler(catch_log)
+    logging.getLogger().addHandler(catch_log_handler)
+    try:
+        s = socket.socket()
+        try:
+            ANY_PORT = 0
+            s.bind((SERVER_ADDRESS, ANY_PORT))
+            address, port = s.getsockname()
+            with ThreadPoolExecutor(max_workers=2) as executor:
+                options = {
+                    "address": SERVER_ADDRESS,
+                    "port": port,
+                    "browser": False,
+                    "detach": False,
+                    "ipv6": False,
+                }
+                future_to_run_web_server = executor.submit(lambda: command_serve.execute(options=options))
+                command_serve.shutdown()
+                result = future_to_run_web_server.result()
+                assert 3 == result
+
+
+                if not sys.platform == 'win32':
```

Comment:
Imagine someone like you stumbles over this in a year or two. This line raises an eyebrow and leaves the puzzling question to be answered: "Why?"

Can you please add a comment here that'll help that someone?

(We need our Nikola community members and want to make life nice for them!)

Reply:
We now see the original Windows exception. We would like to see the `result` being 3. I think we'll get that by setting `show_tracebacks` to `False` in `MyFakeSite`.

It should be `True` in general, which is what `FakeSite` now has, but we need `False` for this particular test, which `MyFakeSite` being the convenient place to get that.

Reply:
ok, so I would set site.show_tracebacks=False inside the test (say right at the top of the test). Is this only needed for windows or for all tests?

I can add some comments later today and can also test this on linux and mac.